hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieHFileDataBlock.java

```diff
@@ -120,19 +122,21 @@ protected byte[] serializeRecords(List<IndexedRecord> records) throws IOExceptio
 
     // Serialize records into bytes
     Map<String, byte[]> sortedRecordsMap = new TreeMap<>();
-    Iterator<IndexedRecord> itr = records.iterator();
+    // Get writer schema
```

Comment:
Is there a possibility write schema is not present? We need to think thru schema evol cases here more carefully and ensure all is good. Thinking out loud.

Reply:
The write schema is also used in HoodieAvroDataBlock. And HoodieAppendHandle#appendDataAndDeleteBlocks has add schema in hearder before writing HoodieDataBlock.